nat-lab/tests/test_batching.py

```diff
@@ -43,82 +42,142 @@ def _generate_setup_parameters(
+    ),
+    (
+        ConnectionTag.DOCKER_UDP_BLOCK_CLIENT_2,
+        AdapterType.LinuxNativeWg,
+    ),
+    (
+        ConnectionTag.DOCKER_INTERNAL_SYMMETRIC_CLIENT,
+        AdapterType.LinuxNativeWg,
+    ),
+    (ConnectionTag.DOCKER_FULLCONE_CLIENT_1, AdapterType.LinuxNativeWg),
+    (ConnectionTag.DOCKER_FULLCONE_CLIENT_2, AdapterType.LinuxNativeWg),
+    (
+        ConnectionTag.MAC_VM,
+        AdapterType.BoringTun,
+    ),
+    (ConnectionTag.WINDOWS_VM_1, AdapterType.WindowsNativeWg),
+    (ConnectionTag.WINDOWS_VM_2, AdapterType.WireguardGo),
+]
+# This test captures histograms of network activity to evaluate the effect of local batching in libtelio.
+# Since only local batching is implemented, no client-generated traffic should occur during the test.
+# External traffic (incoming data) could distort the histograms, and receive-data-triggered batching is
+# not yet supported in libtelio. The test setup is simple: all clients are interconnected and remain idle
+# for an extended period. This idle period allows for a visual observation.
+# Local batching will only have an effect of batching multiple local keepalives into one bundle but will
+# not do anything about syncing the keepalives between the peers.
+
+
+# TODO: Add asserts for local batching
+# TODO: Implement received-data-trigger batching
 @pytest.mark.asyncio
-@pytest.mark.timeout(200)
+@pytest.mark.timeout(TEST_BATCHING_TIMEOUT)
 @pytest.mark.parametrize(
-    "setup_params,misalign_sleep_amount",
+    "setup_params,misalign_sleep_range,capture_duration",
     [
         pytest.param(
             [
-                _generate_setup_parameters(
-                    ConnectionTag.DOCKER_CONE_CLIENT_1,
-                    AdapterType.LinuxNativeWg,
-                    ["stun", "upnp", "local"],
-                    direct_enabled=True,
-                    batching_enabled=True,
-                ),
-                _generate_setup_parameters(
-                    ConnectionTag.DOCKER_CONE_CLIENT_2,
-                    AdapterType.LinuxNativeWg,
-                    ["stun", "upnp", "local"],
-                    direct_enabled=True,
-                    batching_enabled=True,
-                ),
-                _generate_setup_parameters(
-                    ConnectionTag.DOCKER_UPNP_CLIENT_1,
-                    AdapterType.LinuxNativeWg,
-                    ["stun", "upnp", "local"],
-                    direct_enabled=True,
-                    batching_enabled=True,
-                ),
-                _generate_setup_parameters(
-                    ConnectionTag.DOCKER_UPNP_CLIENT_2,
-                    AdapterType.LinuxNativeWg,
-                    ["stun", "upnp", "local"],
-                    direct_enabled=True,
-                    batching_enabled=True,
-                ),
+                _generate_setup_parameters(conn_tag, adapter, False)
+                for conn_tag, adapter in ALL_NODES
+            ],
+            (0, 5),
+            240,
```

Comment:
What do you think about declaring these as constant so it wold be easier from a glance to understand what the numbers mean?

Reply:
That's a good idea, I've extracted those to constants :) :+1:

Reply:
Thanks